File: Parte2/ReservaHotelFront/src/app/models/booking.ts
import { Room } from "./room";

export interface Booking {
    id: number;
    price: number;
    checkIn: Date;
    checkOut: Date;
    roomQtd: number;
    adultsNumber: number;
    childNumber: number;
    
    userId: string;
    roomId: number;
    room?: Room;
}

export interface BookingCreate {
    userId: string;
    price: number;
    checkIn: Date;
    checkOut: Date;
    roomId:number;
    roomQtd:number;
    adultsNumber:number;
    childNumber:number;
}

File: Parte2/ReservaHotelBack/ReservaHotel/Entities/Booking.cs
namespace ReservaHotel.Entities
{
    public class Booking
    {
        public int Id { get; set; }

        public float Price { get; set; }
        public DateTime CheckIn { get; set; }
        public DateTime CheckOut { get; set; }
        public int RoomQtd { get; set; }

        public int AdultsNumber { get; set; }
        public int ChildNumber { get; set; }

        public User User { get; set; }
        public Guid UserId { get; set; }

        public Room Room { get; set; }
        public int RoomId { get; set; }
    }
}
